@kiva/kv-components/vue/KvPagination.vue

```diff
@@ -0,0 +1,198 @@
+import KvMaterialIcon from './KvMaterialIcon.vue';
+
+export default {
+	name: 'KvPagination',
+	components: {
+		KvMaterialIcon,
+	},
+	props: {
+		limit: {
+			type: Number,
+			required: true,
+			validator: (value) => value > 0,
+		},
+		total: {
+			type: Number,
+			required: true,
+			validator: (value) => value >= 0,
+		},
+		offset: {
+			type: Number,
+			default: 0,
+			validator: (value) => value >= 0,
+		},
+		extraPages: {
+			type: Number,
+			default: 3,
+			validator: (value) => value > 0,
+		},
+		scrollToTop: {
+			type: Boolean,
+			default: true,
+		},
+		kvTrackFunction: {
+			type: Function,
+			default: () => {},
+		},
+		trackEventCategory: {
+			type: String,
+			default: '',
+		},
+	},
+	data() {
+		return {
+			mdiChevronLeft,
+			mdiChevronRight,
+		};
+	},
+	computed: {
+		current() {
+			const page = this.offset / this.limit;
+
+			// This component uses a 0-based page index
+			return page < this.totalPages ? page : 0;
+		},
+		totalPages() {
+			return Math.ceil(this.total / this.limit);
+		},
+		numbers() {
+			// If less than the max, there will be no ellipsis, so just return the numbers
+			if (this.totalPages < (this.extraPages + 2)) {
+				return this.range(0, this.totalPages - 1);
+			}
+
+			const numbers = [];
+
+			// Add the 'middle' block of numbers based upon the current page
+			if ([0, 1, 2].includes(this.current)) {
+				numbers.push(...this.range(1, this.extraPages));
+			} else if ([this.totalPages - 3, this.totalPages - 2, this.totalPages - 1]
+				.includes(this.current)) {
+				numbers.push(...this.range(this.totalPages - this.extraPages - 1, this.totalPages - 2));
+			} else {
+				const delta = Math.floor(this.extraPages / 2);
+				numbers.push(...this.range(this.current - delta, this.current + delta));
+			}
+
+			// Add a placeholder for first ellipsis
+			if (numbers[1] !== 2) {
+				numbers.splice(0, 0, -1);
+			}
+
+			// Add a placeholder for second ellipsis
+			const totalNumbers = numbers.length;
+			if (numbers[totalNumbers - 1] !== this.totalPages - 2) {
+				numbers.splice(totalNumbers, 0, -1);
+			}
+
+			// Add first and last pages
+			numbers.unshift(0);
+			numbers.push(this.totalPages - 1);
+
+			return numbers;
+		},
+	},
+	methods: {
+		range(start, end) {
+			return [...Array(end - start + 1)].map((_, n) => n + start);
+		},
+		isCurrent(number) {
+			return number === this.current;
+		},
+		isEllipsis(number) {
+			return number === -1;
+		},
+		linkClass(number) {
+			return { 'tw-text-tertiary': this.isCurrent(number), 'tw-pointer-events-none': this.isCurrent(number) };
+		},
+		pageChange(number) {
+			if (this.scrollToTop && window.scrollTo) {
+				window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+			}
+
+			this.$emit('page-changed', { pageOffset: number * this.limit });
+		},
+		clickPage(number) {
+			this.pageChange(number);
+
+			this.kvTrackFunction(this.trackEventCategory, 'click-page-pager', number + 1);
```

Comment:
Would this be a good time to transition these events to our latest web event guidelines? The actions in particular should be 'click' -- So for this event I would do: 
`this.trackEventCategory, 'click', 'page-pager', number + 1`

Reply:
I guess it depends on if we have reports that depend on these. My guess is we don't, but I don't know for sure...

Reply:
Thank you @eddieferrer That is a good point and I agree we should split the events in the way you have described. `click` or other appropriate action is the 2nd param, followed by a `label` in the 3rd position and then a `value` where appropriate.

Reply:
I think we should use "pagination-number" instead of "page-pager" in this case.

Reply:
@eddieferrer Do you think we should go ahead and duplicate the "number" portion of this as the 5th param?

Reply:
@mcstover I agree with the `pagination-` as the label. For this event maybe: 
`this.trackEventCategory, 'click', 'pagination-number', null, number + 1` would work. But I'm also ok with: 
`this.trackEventCategory, 'click', 'pagination-number', number + 1` just keeping in mind that the 4th param should be a string.